Shared/IISFunctions/Get-IISModules.ps1

```diff
@@ -0,0 +1,159 @@
+                        foreach ($m in $globalModules) {
+                            Write-Verbose "Now processing module: $($m.Name)"
+                            $mInfoObject = [PSCustomObject]@{
+                                Name             = $m.Name
+                                Path             = $null
+                                Signed           = $false
+                                SignatureDetails = $null
+                            }
+
+                            $moduleFilePath = $m.Image
+                            if ($moduleFilePath -match "\%.+\%") {
+                                # Overwrite path when environment variables were found
+                                Write-Verbose "Environment variable found in path: $moduleFilePath"
+                                $moduleFilePath = ConvertCmdEnvironmentVariableToFullPath -PathWithVar $m.Image
+                            }
+
+                            $mInfoObject.Path = $moduleFilePath
+
+                            try {
+                                Write-Verbose "Querying file signing information"
+                                $signature = Get-AuthenticodeSignature -FilePath $moduleFilePath -ErrorAction Stop
+                                # Signature Status Enum Values:
+                                # <0> Valid, <1> UnknownError, <2> NotSigned, <3> HashMismatch,
+                                # <4> NotTrusted, <5> NotSupportedFileFormat, <6> Incompatible,
+                                # https://docs.microsoft.com/dotnet/api/system.management.automation.signaturestatus
+                                if (($signature.Status -ne 1) -and
+                                    ($signature.Status -ne 2) -and
+                                    ($signature.Status -ne 5) -and
+                                    ($signature.Status -ne 6)) {
+                                    Write-Verbose "Signature information found. Status: $($signature.Status)"
+
+                                    $mInfoObject.Signed = $true
+                                    $mDetails = [PSCustomObject]@{
+                                        Signer            = $null
+                                        SignatureStatus   = $signature.Status
+                                        IsMicrosoftSigned = $false
+                                    }
+
+                                    if ($null -ne $signature.SignerCertificate.Subject) {
+                                        Write-Verbose "Signer information found. Subject: $($signature.SignerCertificate.Subject)"
+                                        $mDetails.Signer = $signature.SignerCertificate.Subject.ToString()
+                                        $mDetails.IsMicrosoftSigned = $signature.SignerCertificate.Subject -cmatch "O=Microsoft Corporation, L=Redmond, S=Washington"
+                                    }
+
+                                    $mInfoObject.SignatureDetails = $mDetails
```

Comment:
Create the object here, have the `$signer = $IsMicrosoftSigned = $null` at line 82

Reply:
Done